core/vm/evm.go

```diff
@@ -192,7 +192,7 @@ func (evm *EVM) Call(caller ContractRef, addr common.Address, input []byte, gas
 			// Calling a non existing account, don't do anything, but ping the tracer
 			if debug {
 				if evm.depth == 0 {
-					evm.Config.Tracer.CaptureStart(evm, caller.Address(), addr, false, input, gas, value)
+					evm.Config.Tracer.CaptureStart(caller.Address(), addr, false, input, gas, value)
```

Comment:
(My comment is unrelated to this exact change but quite aligned with context).

Our old instrumentation was instrumenting really early in the call, just after the function definition actually. This has the effect that all state modifications happening was assigned to current call. The best example around this is the `Transfer`. The `Transfer` for a an existing call happens at https://github.com/ethereum/go-ethereum/pull/27629/files#diff-64508d317d86e7a4a5294d76455a5e74148e26883da9e8b4ffc9bbfa3cc8550eR206, so the transfer is "attached" to the "previous" call (or transaction if `depth == 0`).

Is there opening to move that higher to englobe those?

Reply:
It's a call that was executed, it was legitimately called by the smart contract or from the user directly. I don't see good reason why we would not trap all possible errors that happened as part of this call execution.

One of the worst problem for us is the contextuality of state transforming operations that happen before the `CaptureStart`. It prevents attaching easily what happen inside the call to the call itself, here the `Transfer` is happening before the start (agreed, we could move it just before the transfer).

The `create` for example, we do not trace `ErrDepth`, `ErrInsufficientBalance`, `ErrNonceUintOverflow` and `ErrContractAddressCollision` but we do trace `ErrMaxCodeSizeExceeded`, `ErrInvalidCode` and `ErrOutOfGas`.

If feel the `CaptureStart` was placed near the `interpreter.Run`, but logically I do feel that a call starts as soon as `EVM#Call(...)`, `EVM#StaticCall`, etc. is invoked.

 I don't see some errors would be different than others.

Reply:
Capturing the call is now the first thing we do in `evm.Call/Create/...`. So I believe this comment is resolved.